pyanaconda/ui/gui/spokes/subscription.py

```diff
@@ -950,18 +972,62 @@ def _subscription_progress_callback(self, phase):
     def _subscription_error_callback(self, error):
         log.debug("Subscription GUI: registration & attach failed")
         # store the error message
-        self.registration_error = str(error_message)
+        self.registration_error = str(error)
         # even if we fail, we are technically done,
         # so clear the phase
         self.registration_phase = None
         # update registration and subscription parts of the spoke
         self._update_registration_state()
         self._update_subscription_state()
+        # if the error is an instance of multi-org error,
+        # fetch organization list & enable org selection
+        # checkbox
+        if isinstance(error, MultipleOrganizationsError):
+            task_path = self._subscription_module.GetOrganizationDataWithTask()
+            task_proxy = SUBSCRIPTION.get_proxy(task_path)
+            async_run_task(task_proxy, self._process_org_list)
         # re-enable controls, so user can try again
         self.set_registration_controls_sensitive(True)
         # notify hub
         hubQ.send_ready(self.__class__.__name__)
 
+    def _process_org_list(self, task_proxy):
+        """Process org listing for account.
+
+        Called as an async callback of the organization listing runtime task.
+
+        :param task_proxy: a task
+        """
+        # finish the task
+        task_proxy.Finish()
+        # process the organization list
+        org_struct_list = unwrap_variant(task_proxy.GetResult())
+        org_list = OrganizationData.from_structure_list(org_struct_list)
+        # fill the combobox
+        self._select_organization_combobox.remove_all()
+        # also add a placeholder and make it the active item so it is visible
+        self._select_organization_combobox.append("", _("Not Specified"))
```

Comment:
"" is what's passed when no organization is provided, so if nothing is selected the result will be the same, the user will be asked to select an organization again.

As for pre-selecting a value - I've discussed this with @jstodola and he advised against that. 

We have no hints at all about which organization is the correct one & it would be quite likely users would not notice an organization was pre-selected, click Register, possibly resulting in hard to debug issues later on.

Lastly - I don't think the API provides any guarantees about ordering, so that could also cause some problems if the item order is not stable.